test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/MessageProducerClient.java

```diff
@@ -40,32 +41,48 @@ public MessageProducerClient() {
     }
 
     /**
-     * Starts the message producer for a specific topic. Must be called
-     * before any messages can be sent.
+     * Starts the message producer for a specific topic. Must be called before any messages can be
+     * sent.
      *
      * @param topicName the name of the topic to send to
      */
     public void startProducer(final String topicName) {
 
+        if (topicName.equals(this.topicName)) {
+            return;
+        }
+
         try {
-            activeMQConnectionFactory.setBrokerURL(QUEUE_URI);
-            connection = activeMQConnectionFactory.createConnection();
-            connection.start();
+            if (connection == null) {
+                createConnection();
+            }
+
+            if(messageProducer != null) {
```

Comment:
formatting

Reply:
It is formatted - I reformatted before pushing. It just looks a bit weird here

Reply:
if(messageProdu... looks wrong, it looks ok in the IDE ?